### PR TITLE
[OSDEV-1347] Moderation. Implement GET api/v1/moderation-events/{moderation_id}

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-1347](https://opensupplyhub.atlassian.net/browse/OSDEV-1347) Moderation. Implement GET api/v1/moderation-events/{moderation_id}

- Created new GET endpoint api/v1/moderation-events/{moderation_id}
- Added possibility to query search by moderation_id in opensearch
- Updated error handling

[OSDEV-1347]: https://opensupplyhub.atlassian.net/browse/OSDEV-1347?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/django/api/views/v1/moderation_events.py [hunk 1]
```diff
@@ -47,3 +51,34 @@ def list(self, request):
...
+        if not IsRegisteredAndConfirmed().has_permission(request, self):
+            return Response(
+                {
+                    "detail": (
+                        "Only an authorized user "
+                        "can perform this action."
+                    )
+                },
+                status=status.HTTP_401_UNAUTHORIZED
+            )
```
Comment (mazursasha1990) [inline src/django/api/views/v1/moderation_events.py:66]: I suppose the moderation events should be accessible only to moderators.

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

File: src/django/api/views/v1/moderation_events.py (hunk 1 above)
Comment (roman-stolar) [reply src/django/api/views/v1/moderation_events.py:66]: @mazursasha1990 No, there is an requirement that authorized user who want to check the status of his request (moderation event) can get this data by the GET request. As when user publish the POST/PATCH request to moderation queue he get moderation_id (uuid) by this id user could track status.

Comment (mazursasha1990) [reply src/django/api/views/v1/moderation_events.py:66]: Okay.
Thank you.

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

File: src/django/api/views/v1/moderation_events.py [hunk 2]
```diff
@@ -49,3 +52,34 @@ def list(self, request):
...
+        query_params = QueryDict('', mutable=True)
+        query_params.update({'moderation_id': pk})
+        query_body = self.opensearch_query_director.build_query(
+            query_params
+        )
+        response = self.opensearch_service.search_index(
+            OpenSearchIndexNames.MODERATION_EVENTS_INDEX,
+            query_body
+        )
+        return Response(response)
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/django/api/views/v1/moderation_events.py:85]: we didn't use user_id in ModerationEvent model.

File: src/django/api/views/v1/moderation_events.py [hunk 3]
```diff
@@ -49,3 +52,34 @@ def list(self, request):
             query_body
         )
         return Response(response)
+
+    @handle_errors_decorator
+    def retrieve(self, request,  pk=None):
+        if not IsRegisteredAndConfirmed().has_permission(request, self):
```
Comment (VadimKovalenkoSNF) [inline src/django/api/views/v1/moderation_events.py:58]: Can you use `permission_classes = [IsRegisteredAndConfirmed]` here as it works for `list()` method above?

File: src/django/api/views/v1/utils.py [hunk 4]
```diff
@@ -108,3 +109,26 @@ def _wrapped_view(self, request, *args, **kwargs):
         except OpenSearchServiceException as e:
             return handle_opensearch_exception(e)
     return _wrapped_view
+
+
+def is_valid_uuid(uuid_string):
```
Comment (VadimKovalenkoSNF) [inline src/django/api/views/v1/utils.py:114]: This code is already present in `src/django/api/serializers/v1/opensearch_common_validators/moderation_id_validator.py`. Could you use this util function there as well?

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Left few suggestions.

**Review by VadimKovalenkoSNF — APPROVED** [review]

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**